File: app/src/main/java/com/zero/study/ui/dialog/LoadingDialog.java
package com.zero.study.ui.dialog;

import android.app.Dialog;
import android.graphics.Color;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ProgressBar;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

/**
 * @author Admin
 * @date:2024/8/1 11:00
 * @path:com.toolkit.openvpn.dialog.LoadingDialog
 */
public class LoadingDialog extends DialogFragment {

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        ProgressBar progressBar = new ProgressBar(this.getActivity());
        progressBar.setIndeterminate(true);
        return progressBar;
    }

    @Override
    public void onStart() {
        super.onStart();
        if (getDialog() != null && getDialog().getWindow() != null) {
            getDialog().setCanceledOnTouchOutside(true);
            getDialog().getWindow().setBackgroundDrawableResource(android.R.color.transparent);
        }
    }
}

File: library-base/src/main/java/com/zero/base/fragment/LoadingDialog.kt
package com.zero.base.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.Window
import android.widget.ProgressBar
import androidx.fragment.app.DialogFragment
import com.zero.library_base.databinding.BaseLoadingDialogBinding

/**
 * @author Admin
 */
class LoadingDialog : DialogFragment() {
    private var _binding: BaseLoadingDialogBinding? = null
    private val binding get() = _binding!!
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        _binding = BaseLoadingDialogBinding.inflate(inflater, container, false)
        requireDialog().window?.requestFeature(Window.FEATURE_NO_TITLE)
        return binding.root
    }

    override fun onStart() {
        super.onStart()
        if (dialog != null && dialog?.window != null) {
            dialog?.setCanceledOnTouchOutside(false)
            dialog?.window?.setBackgroundDrawableResource(android.R.color.transparent)
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        binding.pbLoading.cancelAnimation()
        _binding = null
    }
}
